Barotrauma/BarotraumaShared/Source/Characters/StatusEffect.cs

```diff
@@ -209,6 +225,80 @@ public virtual bool HasRequiredItems(Entity entity)
             return true;
         }
 
+        public virtual bool HasRequiredConditions(List<ISerializableEntity> targets)
+        {
+            if (!propertyConditionals.Any()) return true;
+            foreach (ISerializableEntity target in targets)
+            {
+                foreach (Tuple<string, string, object> con in propertyConditionals)
+                {
+                    string name = con.Item1;
+                    string op = con.Item2;
+                    object value = con.Item3;
+
+                    SerializableProperty property;
+
+                    if (target == null || target.SerializableProperties == null || !target.SerializableProperties.TryGetValue(name, out property)) continue;
+
+                    Type type = value.GetType();
+                    float? floatValue = null;
+                    if ((type == typeof(float) || type == typeof(int)) && (property.GetValue() is float || property.GetValue() is int))
+                    {
+                        floatValue = Convert.ToSingle(value);
+                    }
+
+                    switch (op)
+                    {
+                        case "==":
+                            if (!(property == value))
```

Comment:
Shouldn't this be comparing the value of the property and not the property itself? Also, you should note that comparing boxed value types won't correctly if you do it this way. Meaning that something like ((object)2) == ((object)2) would equate to false.

https://stackoverflow.com/questions/6205029/comparing-boxed-value-types

Reply:
So I take it I should use .Equals(value) then?

Reply:
Yep, along with a null check to make sure calling Equals doesn't throw an exception.